makefiles/tools/serial.inc.mk

```diff
@@ -5,10 +5,15 @@ ifeq (1,$(MOST_RECENT_PORT))
   endif
   TTY_SELECT_CMD ?= $(RIOTTOOLS)/usb-serial/ttys.py \
                     --most-recent \
-                    --format path \
+                    --format path serial \
                     $(TTY_BOARD_FILTER)
-  PORT_DETECTED := $(shell $(TTY_SELECT_CMD) || echo 'no-tty-detected')
-  PORT ?= $(PORT_DETECTED)
+  TTY_DETECTED := $(shell $(TTY_SELECT_CMD) || echo 'no-tty-detected no-serial-detected')
+  PORT_DETECTED := $(firstword $(TTY_DETECTED))
```

Comment:
Do we ever expect OSes to assign TTY names that contain whitespace, or devices to expose serials that do?

I don't see any direct security concerns, so I'll just leave that as a note and not a blocker.

Reply:
`ttys.py` will quote any output that would contain the "seperation char", which by default is the space. I have to admit that I don't know for sure if `$(firstword "foo bar" "blah blah")` does indeed yield `"foo bar"`, or just `"foo`.